### PR TITLE
RESTWS-716 : Refactored Order Resource to Return An Exception instead of Empty Result with an Order Status Only Parameter

### PR DESCRIPTION
**Pull request by tendomart**
https://issues.openmrs.org/browse/RESTWS-716

@dkayiwa  @mogoodrich  can u please look into this

cc @samuelmale  @wluyima

### CODE REVIEW

**Comment by dkayiwa** [comment]
Did you see the merge conflicts?

**Comment by tendomart** [comment]
@dkayiwa yes seen them later,as it takes long to build.But tried googling not resolved to a solution as yet.

File: omod-1.10/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_10/OrderResource1_10.java [hunk 1]
```diff
@@ -165,13 +169,25 @@ public DelegatingResourceDescription getUpdatableProperties() throws ResourceDoe
...
 	@Override
-	protected PageableResult doSearch(RequestContext context) {
+	protected PageableResult doSearch(RequestContext context) throws ResponseException {
 		String patientUuid = context.getRequest().getParameter("patient");
+		String stat = context.getRequest().getParameter("status");
+		if ((StringUtils.isNotEmpty(patientUuid)) && StringUtils.equals(stat, "any")) {
+             try {
+            	 throw new MissingRequiredIdentifierException("Sorry You did not Specify a Patient #UUID");
+			}
+			catch (MissingRequiredIdentifierException e) {
```
Comment (mogoodrich) [inline omod-1.10/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_10/OrderResource1_10.java]: Why are we catching this?  I believe we just want to throw the exception in this case.

File: omod-1.10/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_10/OrderResource1_10.java [hunk 2]
```diff
@@ -165,13 +169,25 @@ public DelegatingResourceDescription getUpdatableProperties() throws ResourceDoe
 	 * @return all orders for a given patient (possibly filtered by context.type)
 	 */
 	@Override
-	protected PageableResult doSearch(RequestContext context) {
+	protected PageableResult doSearch(RequestContext context) throws ResponseException {
 		String patientUuid = context.getRequest().getParameter("patient");
+		String stat = context.getRequest().getParameter("status");
+		if ((StringUtils.isNotEmpty(patientUuid)) && StringUtils.equals(stat, "any")) {
```
Comment (mogoodrich) [inline omod-1.10/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_10/OrderResource1_10.java]: Shouldn't this be testing if the patientUuid *is* empty?

**Comment by tendomart** [comment]
I refactored and doing a new PR ,i changed  EmptySearchResult.toSimpleObject()

**Pull request closed without merging**